test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/connection/SyncClientConnectionInterruptionTest.java

```diff
@@ -0,0 +1,253 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.exception.AbortedException;
+import software.amazon.awssdk.core.exception.ApiCallAttemptTimeoutException;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.metrics.MetricCollection;
+import software.amazon.awssdk.metrics.MetricPublisher;
+import software.amazon.awssdk.metrics.MetricRecord;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonClient;
+import software.amazon.awssdk.services.protocolrestjson.ProtocolRestJsonClientBuilder;
+import software.amazon.awssdk.services.protocolrestjson.model.AllTypesResponse;
+
+/**
+ * Tests to verify Interruption of Threads while Http Connection is in progress to make sure Resources are released.
+ */
+class SyncClientConnectionInterruptionTest {
+    public static final String SAMPLE_BODY = "{\"StringMember"
+                                             + "\":\"resultString\"}";
+    private final WireMockServer mockServer = new WireMockServer(new WireMockConfiguration()
+                                                                     .bindAddress("localhost")
+                                                                     .dynamicPort());
+    @BeforeEach
+    public void setup() {
+        mockServer.start();
+        stubPostRequest(".*", aResponse(), "{}");
+    }
+
+    @Test
+    void connectionPoolsGetsReusedWhenInterruptedWith_1_MaxConnection() throws Exception {
+        Integer LONG_DELAY = 1500;
+
+        String urlRegex = "/2016-03-11/allTypes";
+        stubPostRequest(urlRegex, aResponse().withFixedDelay(LONG_DELAY), SAMPLE_BODY);
+        SdkHttpClient httpClient = ApacheHttpClient.builder().maxConnections(1).build();
+        ProtocolRestJsonClient client = getClient(httpClient, Duration.ofMillis(2L * LONG_DELAY)).build();
+
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        Future<?> toBeInterruptedFuture = executorService.submit(() -> client.allTypes());
+        unInterruptedSleep(LONG_DELAY - LONG_DELAY / 5);
+        toBeInterruptedFuture.cancel(true);
+        // Make sure thread start the Http connections
+        unInterruptedSleep(50);
+        AllTypesResponse allTypesResponse = client.allTypes();
+        assertThat(allTypesResponse.stringMember()).isEqualTo("resultString");
+        executorService.shutdownNow();
+    }
+
+    @Test
+    void connectionPoolsGetsReusedWhenInterruptedWith_Multiple_MaxConnection() throws Exception {
```

Comment:
Do we actually need this test? It seems we have verified the behavior (connection gets released back to the pool) in `connectionPoolsGetsReusedWhenInterruptedWith_1_MaxConnection`

Reply:
removing this test case